### PR TITLE
File validation updates 

### PR DESCRIPTION
**Pull request by maxinelasp**
# Change Summary
Some updates to file validation

## Overview

Added a new way to create a class using instrument, version, startdate, etc rather than a filename
removed data_dir argument to use config

## Updated Files

- file_validation.py
   - Moved to using config and env var
   - added a new way of creating an instance: `science_file = ScienceFilePath.generate_from_inputs(...)`

## Testing

Fixed tests to align with config

### CODE REVIEW

File: imap_data_access/file_validation.py [hunk 1]
```diff
@@ -68,6 +69,51 @@ def __init__(self, filename: str | Path, data_dir: Path | None = None):
...
+            The end time for the filename
+        version : str
+            The version of the data
+
+        Returns
+        -------
+        str
+            The generated filename
+        """
+        filename = (
```
Comment (tech3371) [inline imap_data_access/file_validation.py:111]: should we check if level is l0, then set extension to be `pkts`?

File: imap_data_access/file_validation.py [hunk 2]
```diff
@@ -68,6 +69,51 @@ def __init__(self, filename: str | Path, data_dir: Path | None = None):
...
+        version: str,
+    ):
+        """Generate a filename from given inputs and return a ScienceFilePath instance.
+
+        This can be used instead of the __init__ method to make a new instance:
+        ```
+        science_file_path = ScienceFilePath.generate_from_inputs("mag", "l0", "test",
+            "20240213", "20240213", "v01-01")
+        full_path = science_file_path.construct_path()
+        ```
```
Comment (bourque) [inline imap_data_access/file_validation.py:89]: Very cool!

For someone who doesn't like python so much, you sure are good at it 😂

File: imap_data_access/file_validation.py [hunk 3]
```diff
@@ -68,6 +69,51 @@ def __init__(self, filename: str | Path, data_dir: Path | None = None):
         if self.error_message:
             raise self.InvalidScienceFileError(f"{self.error_message}")
 
+    @classmethod
+    def generate_from_inputs(
+        cls,
+        instrument: str,
+        level: str,
```
Comment (bourque) [inline imap_data_access/file_validation.py:76]:
```suggestion
        data_level: str,
```
To be consistent with how we are naming this elsewhere

**Review by bourque — APPROVED** [review]
LGTM

File: imap_data_access/file_validation.py (hunk 2 above)
Comment (maxinelasp) [reply imap_data_access/file_validation.py:89]: Flatterer 😆 I guess I take writing good python as a challenge

File: imap_data_access/file_validation.py (hunk 1 above)
Comment (maxinelasp) [reply imap_data_access/file_validation.py:111]: Yes, great catch

**Review by tech3371 — APPROVED** [review]

File: imap_data_access/file_validation.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+# ruff: noqa: PLR0913
 """Methods for managing and validating filenames and filepaths"""
```
Comment (tech3371) [inline imap_data_access/file_validation.py:2]: I found an issue while updating lambda to use this new validation method. Can you update `extract_filename_components`
from this
```
pattern = (
            r"^(?P<mission>imap)_"
            r"(?P<instrument>[^_]+)_"
            r"(?P<datalevel>[^_]+)_"
            r"(?P<descriptor>[^_]+)_"
            r"(?P<startdate>\d{8})_"
            r"(?P<enddate>\d{8})_"
            r"(?P<version>v\d{2}-\d{2})"
            r"\.(?P<extension>cdf|pkts)$"
        )
```
to this
```
pattern = (
            r"^(?P<mission>imap)_"
            r"(?P<instrument>[^_]+)_"
            r"(?P<data_level>[^_]+)_"
            r"(?P<descriptor>[^_]+)_"
            r"(?P<start_date>\d{8})_"
            r"(?P<end_date>\d{8})_"
            r"(?P<version>v\d{2}-\d{2})"
            r"\.(?P<extension>cdf|pkts)$"
        )
```
This way it matches database naming.

File: imap_data_access/file_validation.py [hunk 5]
```diff
@@ -205,7 +255,7 @@ def extract_filename_components(filename: str | Path) -> dict:
         pattern = (
             r"^(?P<mission>imap)_"
             r"(?P<instrument>[^_]+)_"
-            r"(?P<datalevel>[^_]+)_"
```
Comment (tech3371) [inline imap_data_access/file_validation.py:208]: can you also fix `startdate` to `start_date` and `enddate` to `end_date`?

Comment (maxinelasp) [reply imap_data_access/file_validation.py:208]: Sorry, missed that in your first comment!

**Review by tech3371 — APPROVED** [review]

**Pull request merged**